src/briefcase/integrations/subprocess.py

```diff
@@ -385,19 +385,23 @@ def stream_output(self, label, popen_process, stop_func=None):
             name=f"{label} output streamer",
             target=self._stream_output_thread,
             args=(popen_process,),
+            daemon=True,
         )
         try:
             output_streamer.start()
-            if stop_func:
-                while output_streamer.is_alive() and not stop_func():
-                    time.sleep(0.1)
-            else:
-                output_streamer.join()
+            # joining the thread is avoided due to demonstrated
+            # instability of thread interruption via CTRL+C (#809)
+            while not stop_func() and output_streamer.is_alive():
+                sleep(0.1)
         except KeyboardInterrupt:
-            pass  # allow CTRL+C to gracefully stop streaming
+            # allow time for CTRL+C to propagate to the child process
+            sleep(0.25)
         finally:
             self.cleanup(label, popen_process)
-            output_streamer.join()
+            streamer_deadline = time() + 3
+            while output_streamer.is_alive() and time() < streamer_deadline:
+                sleep(0.1)
+            self._log_return_code(popen_process.poll())
 
```

Comment:
This waiting around for the streamer is rather arbitrary.

In any expected case, `is_alive()` will return `False` and we continue on.

It's at least theoretically possible `is_alive()` returns `True`. This should mean the streamer is flooding the console with output or `readline()` is stuck waiting for `EOF`......or something worse.

That said, waiting around forever (like we were) would probably be worse....and moving forward prematurely presents the possibly to mix subprocess output with `briefcase` output....although, at that point, the output wouldn't likely be very valuable anyway.